File: c/printing_quotes.c
#include <stdio.h>
#include <stdlib.h>

int main(void) {
  char quote[50] = {};
  char author[30] = {};

  printf("What is the quote? ");
  fgets(quote, sizeof(quote), stdin);

  printf("Who said it? ");
  fgets(author, sizeof(author), stdin);

  printf("%s says, \"%s\"\n", author, quote);

  return EXIT_SUCCESS;
}

File: zig/printing_quotes.zig
const std = @import("std");

pub fn main() !void {
    const stdin = std.io.getStdIn().reader();
    var buffer: [255]u8 = undefined;

    std.debug.print("What is the quote? ", .{});
    const quote = try stdin.readUntilDelimiter(buffer[0..], '\n');

    std.debug.print("Who said it? ", .{});
    const author = try stdin.readUntilDelimiter(buffer[quote.len..], '\n');

    std.debug.print("{s} says, \"{s}\"", .{ author, quote });
}
